jetstreamext/README.md

```diff
@@ -136,3 +136,86 @@ if err != nil {
 }
 // process msgs
 ```
+
+### Atomic batch publishing
+
+`PublishMsgBatch` and `BatchPublisher` provide atomic batch publishing to JetStream streams with configurable flow control.
+
+In order to use this feature, stream has to be configured with `AllowAtomicPublish` enabled.
+
+> Note: This module requires nats-server v2.12.0 o later.
+
+#### BatchPublisher
+
+`BatchPublisher` allows you to create a publisher that publishes messages in streaming-like fashion, where each message is published individually, but the commit is done for the entire batch.
+It can be configured with options for flow control and supports publish consistency checks.
+A commit is done when the `Commit` method is called, which returns a `BatchAck` containing the results of the publish.
+
+```go
+// Create a stream with batch publishing enabled
+
+// stream has to be created with AllowAtomicPublish enabled
+cfg := jetstream.StreamConfig{
+    Name:               "FOO",
+    Subjects:           []string{"foo.>"},
+    AllowAtomicPublish: true,
+}
+stream, err := js.CreateStream(ctx, cfg)
+if err != nil {
+    // handle error
+}
+
+// Create a batch publisher
+batch, err := jetstreamext.NewBatchPublisher(js)
+if err != nil {
+    // handle error
+}
+
+// Add message to the batch
+err := batch.AddMsg("foo.A", &nats.Msg{
+    Subject: "test.A",
+    Data:    []byte("hello"),
+})
+if err != nil {
+    // handle error
+}
+
+// Commit the batch
+ack, err := batch.Commit(ctx, "test.A", []byte("commit msg"))
```

Comment:
the commit is a normal payload bearing message and technically part of the published batch, why not take a *nats.Msg like AddMsg does?

Reply:
It does,. There's a bit of mix and match in README, but basically you have Add/AddMsg and Commit/CommitMsg to match the pattern used across `nats.go` (Publish/PublishMsg, Request/RequestMsg etc.)

Reply:
Ah, cool thanks!